File: frontend/src/app/models/fieldModel/field.ts
import { Category } from '../categoryModel/category';
import { Liable } from '../liableModel/liable';
import { TnbTax } from '../tnbTaxModel/tnb-tax';

export class Field {
  id!: number;
  surface!: number;
  address!: string;
  category!: Category;
  tnbTaxes!: TnbTax[];
  liable!: Liable;
}

File: backend/ms1/src/main/java/com/example/ms1/entities/Field.java
package com.example.ms1.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Field {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int surface;
    private String address;
    @ManyToOne
    private Category category;
    @OneToMany(mappedBy = "field")
    @JsonIgnore
    private List<TnbTax> tnbTaxes;
    @ManyToOne
    private Liable liable;

}
